controller/konnect/ops_kongconsumer.go

```diff
@@ -0,0 +1,195 @@
+			ControlPlaneID: cp.Status.ID,
+			ConsumerID:     consumer.GetKonnectStatus().GetKonnectID(),
+			Consumer:       kongConsumerToSDKConsumerInput(consumer),
+		},
+	)
+
+	// TODO: handle already exists
+	// Can't adopt it as it will cause conflicts between the controller
+	// that created that entity and already manages it, hm
+	if errWrapped := wrapErrIfKonnectOpFailed[configurationv1.KongConsumer](err, UpdateOp, consumer); errWrapped != nil {
+		k8sutils.SetCondition(
+			k8sutils.NewConditionWithGeneration(
+				KonnectEntityProgrammedConditionType,
+				metav1.ConditionFalse,
+				"FailedToCreate",
+				errWrapped.Error(),
+				consumer.GetGeneration(),
+			),
+			consumer,
+		)
+		return errWrapped
+	}
+
+	consumer.Status.Konnect.SetKonnectID(*resp.Consumer.ID)
+	consumer.Status.Konnect.SetControlPlaneID(cp.Status.ID)
+	k8sutils.SetCondition(
+		k8sutils.NewConditionWithGeneration(
+			KonnectEntityProgrammedConditionType,
+			metav1.ConditionTrue,
+			KonnectEntityProgrammedReason,
+			"",
+			consumer.GetGeneration(),
+		),
+		consumer,
+	)
+
+	return nil
+}
+
+func deleteConsumer(
+	ctx context.Context,
+	sdk *sdkkonnectgo.SDK,
+	consumer *configurationv1.KongConsumer,
+) error {
+	id := consumer.Status.Konnect.GetKonnectID()
+	if id == "" {
+		return fmt.Errorf("can't remove %T without a Konnect ID", consumer)
+	}
+
+	_, err := sdk.Consumers.DeleteConsumer(ctx, consumer.Status.Konnect.ControlPlaneID, id)
+	if errWrapped := wrapErrIfKonnectOpFailed[configurationv1.KongConsumer](err, DeleteOp, consumer); errWrapped != nil {
+		// Consumer delete operation returns an SDKError instead of a NotFoundError.
+		var sdkError *sdkerrors.SDKError
+		if errors.As(errWrapped, &sdkError) && sdkError.StatusCode == 404 {
+			ctrllog.FromContext(ctx).
+				Info("entity not found in Konnect, skipping delete",
+					"op", DeleteOp, "type", consumer.GetTypeName(), "id", id,
+				)
+			return nil
+		}
+		return FailedKonnectOpError[configurationv1.KongConsumer]{
+			Op:  DeleteOp,
+			Err: errWrapped,
+		}
+	}
+
+	return nil
+}
+
+func kongConsumerToSDKConsumerInput(
+	consumer *configurationv1.KongConsumer,
+) sdkkonnectgocomp.ConsumerInput {
+	return sdkkonnectgocomp.ConsumerInput{
+		CustomID: &consumer.CustomID,
+		Tags:     ExtractUserTags(consumer),
+		Username: &consumer.Username,
+	}
+}
+
+// ExtractUserTags extracts a set of tags from a comma-separated string.
+// Copy pasted from: https://github.com/Kong/kubernetes-ingress-controller/blob/eb80ec2c58f4d53f8c6d7c997bcfb1f334b801e1/internal/annotations/annotations.go#L407-L416
+func ExtractUserTags(obj metav1.Object) []string {
+	anns := obj.GetAnnotations()
+	val := anns[AnnotationPrefix+UserTagKey]
+	// If the annotation is not present, the map provides an empty value,
+	// and splitting that will create a slice containing a single empty string tag.
+	// These aren't valid, hence this special case.
+	if len(val) == 0 {
+		return []string{}
+	}
+	return strings.Split(val, ",")
+}
```

Comment:
We could move this function to `kubernetes-configuration`, WDYT?

Reply:
https://github.com/Kong/kubernetes-configuration/pull/45